charmhelpers/fetch/ubuntu.py

```diff
@@ -760,9 +765,16 @@ def _add_apt_repository(spec):
     :param spec: the parameter to pass to add_apt_repository
     :type spec: str
     """
+    series = get_distrib_codename()
     if '{series}' in spec:
-        series = get_distrib_codename()
         spec = spec.replace('{series}', series)
+    cmd = ['add-apt-repository', '--yes']
+    if (spec.strip().startswith('deb')
+            and CompareHostReleases(series) >= 'jammy'):
+        # if starts with deb, just add the source.list as per lp:2017014, only
+        # in jammy or newer
+        cmd.append('-S')
+    cmd.append(spec)
     _run_with_retries(['add-apt-repository', '--yes', spec],
                       cmd_env=env_proxy_settings(['https', 'http', 'no_proxy'])
                       )
```

Comment:
This code doesn't actually use `cmd` in `_run_with_retries(...)`?  I'm guessing this isn't the actual code that is being used?

If 'deb' needs specially handling at jammy+ then the `_mapping` further up the code could probably pick it out separately from `http:`, `https:`, and `ppa:` to make it obvious that just `deb` is being handled specifically.

Reply:
indeed, pushed things in a rush to provide something, will fix and push it
AFAIK, only deb for private ppa but haven't tested other cases